mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -0,0 +1,251 @@
+        f"Argument {arg_name} is unsupported for models in the Unity Catalog.",
+    ]
+    if message is not None:
+        messages.append(message)
+    messages.append("See the user guide for more information")
+    raise MlflowException(" ".join(messages))
+
+
+def _raise_unsupported_method(method, message=None):
+    messages = [
+        f"Method {method} is unsupported for models in the Unity Catalog.",
+    ]
+    if message is not None:
+        messages.append(message)
+    messages.append("See the user guide for more information")
+    raise MlflowException(" ".join(messages))
+
+
+def _get_response_from_method(method):
+    method_to_response = {
+        CreateRegisteredModelRequest: CreateRegisteredModelResponse,
+        UpdateRegisteredModelRequest: UpdateRegisteredModelResponse,
+        DeleteRegisteredModelRequest: DeleteRegisteredModelResponse,
+        CreateModelVersionRequest: CreateModelVersionResponse,
+        FinalizeModelVersionRequest: FinalizeModelVersionResponse,
+        UpdateModelVersionRequest: UpdateModelVersionResponse,
+        DeleteModelVersionRequest: DeleteModelVersionResponse,
+        GetModelVersionDownloadUriRequest: GetModelVersionDownloadUriResponse,
+        SearchModelVersionsRequest: SearchModelVersionsResponse,
+        GetRegisteredModelRequest: GetRegisteredModelResponse,
+        GetModelVersionRequest: GetModelVersionResponse,
+        SearchRegisteredModelsRequest: SearchRegisteredModelsResponse,
+        # pylint: disable=line-too-long
+        GenerateTemporaryModelVersionCredentialsRequest: GenerateTemporaryModelVersionCredentialsResponse,
+    }
+    return method_to_response[method]()
+
+
+def _get_all_endpoints_from_method(method):
+    return _METHOD_TO_ALL_INFO[method]
+
+
+def _get_endpoint_from_method(method):
+    return _METHOD_TO_INFO[method]
+
+
+# TODO: re-enable the abstract-method check in a follow-up PR. It's disabled for now because
+# we haven't yet implemented required model version CRUD APIs in this class
+class UcModelRegistryStore(BaseRestStore):  # pylint: disable=abstract-method
+    """
+    Note:: Experimental: This entity may change or be removed in a future release without warning.
+    Client for a remote model registry server accessed via REST API calls
+
+    :param get_host_creds: Method to be invoked prior to every REST request to get the
+      :py:class:`mlflow.rest_utils.MlflowHostCreds` for the request. Note that this
+      is a function so that we can obtain fresh credentials in the case of expiry.
+    """
+
+    def __init__(self, get_host_creds):
+        super().__init__(
+            get_host_creds,
+            get_response_from_method=_get_response_from_method,
+            get_endpoint_from_method=_get_endpoint_from_method,
+            get_all_endpoints_from_method=_get_all_endpoints_from_method,
+        )
+
+    # CRUD API for RegisteredModel objects
+
+    def create_registered_model(self, name, tags=None, description=None):
```

Comment:
is `tags` a placeholder for future work? It's not used in the method.

Reply:
(if not supported, should there be a `_require_arg_unspecified()` call here to let users know that it's currently not supported?)

Reply:
Thanks for the catch! Yes, we should be throwing that `tags` are not supported yet